src/components/controllers/doc-page.jsx

```diff
@@ -53,9 +51,8 @@ export function DocLayout() {
 
 function OldDocsWarning() {
 	const { name, version } = useRoute().params;
-	const { path } = useLocation();
 
-	if (!isDocPage(path) || version === LATEST_MAJOR) {
+	if (version === LATEST_MAJOR) {
```

Comment:
This was leading to a flash of the banner when navigating from (say) `/` -> `/guide/v10/getting-started`, as while the `path` instantly changes, `useRoute()` had a delay in delivering the route props as the router had to switch from `Page` -> `DocLayout` to `DocPage` -> `DocLayout`